### ISSUE TITLE
Slow admin interface on music files (?)

### ISSUE THREAD

**Issue by gammaw** [AzuraCast/AzuraCast#2308]
Hi,

didn't the Music Files view and filtering get slower? We have the same server (w/ 2GB memory) with the same amount of files (around 10GB, <150 files). It takes seconds to load all files (same for filtering).  

Anybody have a similar issue? 

Installation: v0.9.8, #c72be08 (2019-12-15 8:53)

PS: I was sure that I had posted this one some days ago but I couldn't find it. Sorry if it turns out to be a duplicate.

**Comment by SlvrEagle23** [comment]
@gammaw We haven't changed anything about the music files section in a while...it really shouldn't be any slower. If anything, I've been working on several of the reports to make them _faster_ and more memory-optimized to avoid problems on low-cost hardware.

**Comment by gammaw** [comment]
@SlvrEagle23 I see. What exactly is happening when the list items are grey? Is it only load on the client or is there communication between server and client going on? This is where we see most delay.

**Comment by SlvrEagle23** [comment]
@gammaw You mean when you've done something to them and it's reloading? When they turn grey, that's when it's making the request back to the server to "relist" the directory with whatever changes you made applied. That request invalidates the previous cache (since changes were made to the directory structure) so it has to rebuild the cache...which isn't particularly slow but you'd notice it if the directory structure was particularly large.

**Comment by gammaw** [comment]
@SlvrEagle23 I see. Thanks. Strangely, it seems to vary a lot how fast this operation is on our server (e.g., waiting on a filter result) although we don't have any other load on that server (beyond the Azuracast instance). I don't see any particular pattern (e.g., slow display in admin interface when lots of listeners). Btw, what do you mean by "directory structure was particularly large"? In the `media` folder, music files are stored "flat" (all in that folder). Like I said we have around 150 files, which shouldn't be too many.

**Comment by gammaw** [comment]
@SlvrEagle23 do you think this issue can be related to RAM shortage (see: https://github.com/AzuraCast/AzuraCast/issues/2342)?

**Comment by SC2Mitch** [comment]
This could be related to memory issues @gammaw yes

**Comment by SlvrEagle23** [comment]
@gammaw It's entirely possible that you could be running out of RAM and causing the disk to swap out, which can make the numerous database queries involved in the media manager much slower.

The overall performance of the page also has to do with how many records you have visible on a single page. Some processing only happens for the visible records, not all of the ones in the folder, so using a lower per-page number can help reduce issues sometimes.

**Comment by gammaw** [comment]
@SC2Mitch @SlvrEagle23 thanks, good to know, we'll try to adjust there. To be honest, we don't see too much speed-up after upgrading from 2 to 3GB, which brings me to my question in the other thread: would Azuracast use all memory it gets from the OS? If we upgrade to 4GB would the utilization still be almost 100%? Again, with 2GB and 3GB we're seeing pretty much the same amount of free memory (around 200K).

**Comment by SlvrEagle23** [comment]
@gammaw It's hard to answer authoritatively without actually seeing your setup, what kind of resources it's using and how it's allocated. This could be an issue of applications just reserving more RAM than they need and keeping it "handy" (MariaDB in particular likes to do this, it's a bit like a goldfish in that it expands to fill whatever size it's given).

When you say listing the directories is slow, how slow are we talking about, and how many records are visible per page?

**Comment by gammaw** [comment]
@SlvrEagle23 just checked and it was 50 entries per page, now I changed it to 10 and it looks faster. The speed varies, sometimes it's instant, sometimes it's seconds. It's also our usage why the response time is important for us: we upload new music, edit & save and assign a playlist. This workflow includes two listing and two searches. We submitted once a feature request on upvote where the playlist can be assigned in the edit window. That would halve the number of listings/searches. But it would be just an enhancement, it's fair how it is now too and we're very happy with Azura in general :) Now with 3GB RAM things look smoother, I can imagine that with 2GB the system was crawling. We'll see it on the long run. Since the upgrade we didn't have any outage (unexpected, undeterministic error etc.). We can still upgrade to 4GB.
